R/git.R

```diff
@@ -0,0 +1,37 @@
+git_run <- function(args, repo = NULL, check = FALSE) {
+  git <- sys_which("git")
+  if (!is.null(repo)) {
+    args <- c("-C", repo, args)
+  }
+  res <- system3(git, args)
+  if (check && !res$success) {
+    stop(sprintf("Error code %d running command:\n%s",
+                 res$code, paste0("  > ", res$output, collapse = "\n")))
+  }
+  res
+}
+
+
+git_ref_to_sha <- function(ref, repo = NULL, check = FALSE) {
+  assert_scalar_character(ref)
+  res <- git_run(c("rev-parse", ref), repo = repo, check = FALSE)
+  if (res$success) {
+    res$output
+  } else if (check) {
+    stop(sprintf("Git reference '%s' not found", ref), call. = FALSE)
+  } else {
+    NA_character_
+  }
+}
+
+
+git_clone_depth_1 <- function(origin, ref) {
```

Comment:
Looks like outpack is saving the sha and the branch. What the runner displays will be coming from the database from what we discussed on Monday. So I think that is fine, as long as we're happy that what we show in the UI will be different from what is in the metadata for the branch. The sha will be the same though, so perhaps that is enough.

Reply:
Ok think I'm with you now, so what we have now in this PR is fine for listing packages (just duping the logic from orderly2 instead of doing shallow checkouts).

I was hoping this function could be re-used for doing the running as we'd planned just taking a sha as params for the run endpoint. But sounds like we want to branch to, so to sum up. The runner will
1. Take a branch name, a commit sha, report to be run, parameters for the report
1. Do a shallow clone of only the report source (entire src dir? or just src/<report name>) to be run into a new directory from the sha, set branch name
1. Run the report via rrq with `separate_process = TRUE`. Use `root_src` to point to cloned src directory. 
1. Return the ID of the job